assembly/src/parsers/io_ops/env_ops.rs

```diff
@@ -1,51 +1,59 @@
-/// stack.
-/// - `push.env.sdepth` pushes the current depth of the stack onto the top of the stack, which is
-/// handled directly by the `SDEPTH` operation.
+/// Appends `locaddr.i` operation to the span block to push the absolute address of the local
+/// variable at index `i` onto the stack.
 ///
 /// # Errors
 ///
-/// This function expects a valid assembly environment op that specifies the environment input to
-/// be handled. It will return an error if the assembly instruction is malformed or the environment
-/// input is unrecognized.
-pub fn parse_push_env(
+/// It will return an error if the assembly instruction is malformed or it has inappropriate
+/// parameter value according to the number of local variables of the procedurethe
+pub fn parse_locaddr(
     span_ops: &mut Vec<Operation>,
     op: &Token,
     num_proc_locals: u32,
 ) -> Result<(), AssemblyError> {
-    validate_operation!(op, "push.env.locaddr|sdepth");
-
-    // update the span block
-    match op.parts()[2] {
-        "locaddr" => {
-            if num_proc_locals == 0 {
-                return Err(AssemblyError::invalid_op_with_reason(
-                    op,
-                    "no procedure locals available in current context",
-                ));
-            }
-            validate_operation!(@only_params op, "push.env.locaddr", 1);
-            let index = parse_u32_param(op, 3, 0, num_proc_locals - 1)?;
-
-            push_value(span_ops, -Felt::new(index as u64));
-            span_ops.push(Operation::FmpAdd);
-        }
-        "sdepth" => {
-            validate_operation!(@only_params op, "push.env.sdepth", 0);
-            span_ops.push(Operation::SDepth);
-        }
-        _ => return Err(AssemblyError::invalid_op(op)),
+    if op.parts()[0] != "locaddr" {
+        return Err(AssemblyError::unexpected_token(op, "locaddr"));
+    }
+    if op.num_parts() < 2 {
+        return Err(AssemblyError::missing_param(op));
+    }
+    if op.num_parts() > 2 {
+        return Err(AssemblyError::extra_param(op));
+    }
+
+    if num_proc_locals == 0 {
+        return Err(AssemblyError::invalid_op_with_reason(
+            op,
+            "no procedure locals available in current context",
+        ));
+    }
+    let index = parse_u32_param(op, 1, 0, num_proc_locals - 1)?;
+
+    push_value(span_ops, -Felt::from(index));
+    span_ops.push(Operation::FmpAdd);
+
+    Ok(())
+}
+
+/// Appends `sdepth` operation to the current span block to push the current depth of the stack
+/// onto the top of the stack. `sdepth` is handled directly by the `SDEPTH` operation.
+///
+/// # Errors
+///
+/// It will return an error if the assembly instruction is malformed.
+pub fn parse_sdepth(span_ops: &mut Vec<Operation>, op: &Token) -> Result<(), AssemblyError> {
+    if op.num_parts() > 1 {
+        return Err(AssemblyError::extra_param(op));
     }
+    if op.parts()[0] != "sdepth" {
+        return Err(AssemblyError::unexpected_token(op, "sdepth"));
+    }
```

Comment:
Let's replace this check with `debug_assert_eq` as described in the other comment. Also, I'm move it up to be before the `num_parts` check.